plat/kvm/x86/multiboot.c

```diff
@@ -133,33 +124,21 @@ void multiboot_entry(struct lcpu *lcpu, struct multiboot_info *mi)
 			if (end <= start)
 				continue;
 
+			mrd.pbase = start;
+			mrd.vbase = start; /* 1:1 mapping */
+			mrd.len   = end - start;
+
 			if (m->type == MULTIBOOT_MEMORY_AVAILABLE) {
 				mrd.type  = UKPLAT_MEMRT_FREE;
 				mrd.flags = UKPLAT_MEMRF_READ |
 					    UKPLAT_MEMRF_WRITE;
 
-				for (i = 0; i < bi->mrds.count; i++) {
-					ukplat_memregion_get(i, &mrdp);
-					if (!mrd_overlap(start, end, mrdp))
-						continue;
-
-					if (!mrdp->type)
-						continue;
-
-					if (start < mrdp->pbase) {
-						mrd.pbase = start;
-						mrd.vbase = start; /* 1:1 map */
-						mrd.len   = mrdp->pbase - start;
-
-						if (mrd.len >= PAGE_SIZE)
-							mrd_insert(bi, &mrd);
-					}
-
-					start = mrdp->pbase + mrdp->len;
-				}
-
-				if (end - start < PAGE_SIZE)
-					continue;
+				rc = ukplat_memregion_list_insert_split_phys(
+				    &bi->mrds, &mrd, __PAGE_SIZE);
+				if (rc < 0)
+					multiboot_crash("unable to add region",
```

Comment:
Just a nitpick, but perhaps it would look nicer if all crash messages would begin with a capital letter, instead of having messages that begin with capital letters as well as messages that begin with lowercase letters. I personally find it inconsistent.